src/rdkafka_conf.c

```diff
@@ -1451,24 +1485,32 @@ rd_kafka_conf_res_t rd_kafka_conf_set (rd_kafka_conf_t *conf,
                                        const char *name,
                                        const char *value,
                                        char *errstr, size_t errstr_size) {
-        rd_kafka_conf_res_t res;
-
-        res = rd_kafka_anyconf_set(_RK_GLOBAL, conf, name, value,
-                                   errstr, errstr_size);
+        rd_kafka_conf_res_t res = rd_kafka_anyconf_set(_RK_GLOBAL, conf, name, value,
+                                                       errstr, errstr_size);
         if (res != RD_KAFKA_CONF_UNKNOWN)
-                return res;
-
-        /* Fallthru:
+            return res;
+        
+        /* Fallthrough:
          * If the global property was unknown, try setting it on the
          * default topic config. */
         if (!conf->topic_conf) {
-                /* Create topic config, might be over-written by application
-                 * later. */
-                conf->topic_conf = rd_kafka_topic_conf_new();
+            /* Create topic config, might be over-written by application
+             * later. */
+            conf->topic_conf = rd_kafka_topic_conf_new();
+            
+            /* Update the scope to match that of the handle configuration */
+            conf->topic_conf->scope |= (conf->scope & _RK_PC_MASK);
         }
-
-        return rd_kafka_topic_conf_set(conf->topic_conf, name, value,
-                                       errstr, errstr_size);
+        
+        rd_kafka_conf_res_t rc = rd_kafka_topic_conf_set(conf->topic_conf,
```

Comment:
coding guidelines:
 * no inline variable definitions, they must go at the beginning of the block